spec/eof.md

```diff
@@ -369,9 +369,10 @@ During scanning, for each instruction:
 2. Determine the effect the instruction has on the operand stack:
    1. Check if the recorded stack height bounds satisfy the instruction requirements. Specifically:
       - for `CALLF` the following must hold: `stack_height_min >= types[target_section_index].inputs`,
-      - for `RETF` the following must hold: `stack_height_max == stack_height_min == types[current_code_index].outputs`,
+      - for `RETF` the following must hold: `stack_height_max == stack_height_min == types[current_section_index].outputs`,
       - Stack validation of `JUMPF` depends on "non-returning" status of target section
-         - `JUMPF` into returning section (can be only from returning section): `stack_height_min == stack_height_max == type[current_section_index].outputs + type[target_section_index].inputs - type[target_section_index].outputs`
+         - `JUMPF` into returning section (can be only from returning section): `stack_height_min == stack_height_max == type[current_section_index].outputs + type[target_section_index].inputs - type[target_section_index].outputs`.
+           **NOTE**: This means `JUMPF` must ensure target's inputs, must ensure its own outputs (to return control), but must take into account that the target section may leave extra output items after it finishes, so it may need that many fewer items on the operand stack
```

Comment:
"must ensure" is duplicated here

Consider taking explanation from the EIP:
> This means that target section can output less stack elements than the original code section called by the top element on the return stack, if the current code section leaves the delta `type[current_section_index].outputs - type[target_section_index].outputs` element(s) on the stack.

Reply:
Generally EIPs have more explanations than mega-spec, and I'm not sure we have to repeat everything here.

Reply:
I updated the wording a little bit. LMK if you prefer the EIP one after all